doc/experiment.rst

```diff
@@ -1,327 +1,418 @@
+.. compound::
+  In the following subsections, we provide an example of using SmartSim to automate the
+  deployment of an HPC workload consisting of a ``Model`` and standard ``Orchestrator``.
+
+  Continue to the example to:
+
+  .. list-table:: Experiment example contents
+   :widths: auto
+   :header-rows: 1
+
+   * - Initialize
+     - Start
+     - Stop
+   * - a workflow (``Experiment``)
+     - the in-memory database (``Orchestrator``)
+     - the in-memory database (``Orchestrator``)
+   * - a in-memory database (``Orchestrator``)
+     - the workload (``Model``)
+     - 
+   * - a workload (``Model``)
+     - 
+     - 
+
+Initialize
+----------
+.. compound::
+  To create a workflow, we *initialize* an ``Experiment`` object
+  once at the beginning of the Python driver script.
+  To create an Experiment, we specify a name
+  and the system launcher with which all entities will be executed.
+  Here, we are running the example on a Slurm machine and as such will
+  set the `launcher` argument to `slurm`.
+
+  .. code-block:: python
+
+      from smartsim import Experiment
+      from smartsim.log import get_logger
+
+      # Initialize an Experiment
+      exp = Experiment("name-of-experiment", launcher="slurm")
+      # Initialize a SmartSim logger
+      smartsim_logger = get_logger("tutorial-experiment")
+
+  We also initialize a SmartSim logger. We will use the logger throughout the experiment
+  to monitor the entities.
+
+.. compound::
+  Next, we will launch a SmartSim in-memory database called an ``Orchestrator``.
+  To *initialize* an ``Orchestrator`` object, use the ``Experiment.create_database()``
+  function. We will create a single-sharded database and therefore will set
+  the argument `db_nodes` to 1. SmartSim will assign a `port` to the database
+  and attempt to detect your machine's interface if values are not provided to the ``Experiment.create_database()`` factory method.
+
+  .. code-block:: python
+
+      # Initialize an Orchestrator
+      database = exp.create_database(db_nodes=1)
+      # Create an output directory
+      exp.generate(database)
+
+  We use the ``Experiment.generate()`` function to create an
+  output directory for the database log files.
+
+.. compound::
+  Before invoking the factory method to create a ``Model``, we must
+  first create a ``RunSettings`` object which holds the information needed to executed the ``Model``
+  on the system. The ``RunSettings`` object is initialized using the
+  ``Experiment.create_run_settings()`` factory method. In this factory method,
+  we specify the executable to run and the arguments to pass to
+  the executable.
+
+  The example ``Model`` is a simple `Hello World` program
+  that echos `Hello World` to stdout.
+
+  .. code-block:: python
+
+      settings = exp.create_run_settings("echo", exe_args="Hello World")
+      model = exp.create_model("hello_world", settings)
+
+  Notice above we are creating the ``Model`` through the ``Experiment.create_model()``
```

Comment:
Slight rewording to:

  After creating the ``RunSettings`` object, the ``Model`` object can be created and initialized using the ``RunSettings`` object via the ``Experiment.create_model()``
  function. In the ``Model`` factory method, the ``Model`` `name` and the ``RunSettings`` object are provided as input parameters.